gitlab/resource_gitlab_environment.go

```diff
@@ -0,0 +1,132 @@
+package gitlab
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabEnvironment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabEnvironmentCreate,
+		Read:   resourceGitlabEnvironmentRead,
+		Update: resourceGitlabEnvironmentUpdate,
+		Delete: resourceGitlabEnvironmentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"external_url": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"slug": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceGitlabEnvironmentCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	project := d.Get("project").(string)
+	options := &gitlab.CreateEnvironmentOptions{
+		Name:        gitlab.String(d.Get("name").(string)),
+		ExternalURL: gitlab.String(d.Get("external_url").(string)),
+	}
+
+	environment, _, err := client.Environments.CreateEnvironment(project, options)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(strconv.Itoa(environment.ID))
+
+	return resourceGitlabEnvironmentRead(d, meta)
+}
+
+func resourceGitlabEnvironmentRead(d *schema.ResourceData, meta interface{}) error {
```

Comment:
To help future-proof this resource, can you change your resource ID to include all the bits needed to do a read? Maybe `project_id:environment_id`? This way someone can come in and write an `Import` function later on.

Reply:
Could you make this change to the repository file resource too?

Reply:
Could you make this change to the repository file resource too?

Reply:
This has been actioned for environment in: cbf35b79 I am not sure how I am going to address `file` just yet, I want to think on that a bit. I will resolve this though.